### PR TITLE
cleanup: remove package csi duplicated error log

### PR DESCRIPTION
**Pull request by beautytiger**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
cleanup csi package error message, make it more clean to end-user.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/volume/csi/csi_plugin.go [hunk 1]
```diff
@@ -124,7 +124,6 @@ func (h *RegistrationHandler) ValidatePlugin(pluginName string, endpoint string,
 		// CSI 1.x drivers should use the /var/lib/kubelet/plugins_registry
 		if !isDeprecatedSocketDirAllowed(versions) {
 			err := fmt.Errorf("socket for CSI driver %q versions %v was found in a deprecated dir. Drivers implementing CSI 1.x+ must use the new dir", pluginName, versions)
-			klog.Error(err)
 			return err
```
Comment (msau42) [inline pkg/volume/csi/csi_plugin.go]: We can return the errors in this file directly without allocating it first

File: pkg/volume/csi/csi_util.go [hunk 2]
```diff
@@ -44,8 +45,7 @@ func getCredentialsFromSecret(k8s kubernetes.Interface, secretRef *api.SecretRef
 	credentials := map[string]string{}
 	secret, err := k8s.CoreV1().Secrets(secretRef.Namespace).Get(secretRef.Name, meta.GetOptions{})
 	if err != nil {
-		klog.Errorf("failed to find the secret %s in the namespace %s with error: %v\n", secretRef.Name, secretRef.Namespace, err)
-		return credentials, err
+		return credentials, errors.New(log("failed to find the secret %s in the namespace %s with error: %v\n", secretRef.Name, secretRef.Namespace, err))
```
Comment (msau42) [inline pkg/volume/csi/csi_util.go]: remove newline from this string

File: pkg/volume/csi/csi_plugin.go [hunk 3]
```diff
@@ -123,9 +123,7 @@ func (h *RegistrationHandler) ValidatePlugin(pluginName string, endpoint string,
 		// The deprecated dir will only be allowed for a whitelisted set of old versions.
 		// CSI 1.x drivers should use the /var/lib/kubelet/plugins_registry
 		if !isDeprecatedSocketDirAllowed(versions) {
-			err := fmt.Errorf("socket for CSI driver %q versions %v was found in a deprecated dir. Drivers implementing CSI 1.x+ must use the new dir", pluginName, versions)
-			klog.Error(err)
-			return err
+			return fmt.Errorf("socket for CSI driver %q versions %v was found in a deprecated dir. Drivers implementing CSI 1.x+ must use the new dir", pluginName, versions)
```
Comment (msau42) [inline pkg/volume/csi/csi_plugin.go]: When should we use `fmt.Errorf` vs `errors.New`?

File: pkg/volume/csi/csi_util.go (hunk 2 above)
Comment (ethan-daocloud) [reply pkg/volume/csi/csi_util.go]: Good catch, got it. Thank you!

File: pkg/volume/csi/csi_plugin.go (hunk 3 above)
Comment (ethan-daocloud) [reply pkg/volume/csi/csi_plugin.go]: I think it basiclly the same thing, when we need to format the output string,  fmt.Errorf is better in use, when it just raw string, erros.New is ok. fmt.Errof internally called errors.New after formatting.

File: pkg/volume/csi/csi_plugin.go [hunk 4]
```diff
@@ -180,23 +178,20 @@ func (h *RegistrationHandler) RegisterPlugin(pluginName string, endpoint string,
 func (h *RegistrationHandler) validateVersions(callerName, pluginName string, endpoint string, versions []string) (*utilversion.Version, error) {
 	if len(versions) == 0 {
 		err := fmt.Errorf("%s for CSI driver %q failed. Plugin returned an empty list for supported versions", callerName, pluginName)
```
[the remark this answers is not in the document]
Comment (ethan-daocloud) [reply pkg/volume/csi/csi_plugin.go]: Got it, thank you.

File: pkg/volume/csi/csi_plugin.go (hunk 3 above)
Comment (ethan-daocloud) [reply pkg/volume/csi/csi_plugin.go]: errors.New(log()) = fixed-csi-info-string + fmt.Errorf(). 
Shoud we use errors.New(log()) for uniform format?

Comment (msau42) [reply pkg/volume/csi/csi_plugin.go]: Yes I think that is a good idea

File: pkg/volume/csi/csi_plugin.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (msau42) [reply pkg/volume/csi/csi_plugin.go]: sorry I mean we can just do `return nil, errors.New()` directly

[the remark this answers is not in the document]
Comment (ethan-daocloud) [reply pkg/volume/csi/csi_plugin.go]: :)

File: pkg/volume/csi/csi_plugin.go (hunk 3 above)
Comment (ethan-daocloud) [reply pkg/volume/csi/csi_plugin.go]: Got it, thank you for your suggestion.

**Comment by ethan-daocloud** [comment]
retest

**Pull request merged**